### PR TITLE
 RPC144-fix reentrancy

### PR DESCRIPTION
**Pull request by SKonstantinS**

### CODE REVIEW

File: exchange-v2/contracts/ExchangeV2Core.sol [hunk 1]
```diff
@@ -23,6 +23,9 @@ abstract contract ExchangeV2Core is Initializable, OwnableUpgradeable, AssetMatc
     //on-chain orders
     mapping(bytes32 => OrderAndFee) public onChainOrders;
 
+    //prevent reentrancy
+    bool internal locked;
```
Comment (disenotov) [inline exchange-v2/contracts/ExchangeV2Core.sol:27]: can we fix reentrancy without adding new variable? for example setting internal variables before transfer

Comment (disenotov) [reply exchange-v2/contracts/ExchangeV2Core.sol:27]: and in "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol" this variable is initialized, but here it's not. is it ok? won't it break something?

Comment (SKonstantinS) [reply exchange-v2/contracts/ExchangeV2Core.sol]: done, delete modifier

**Comment by SKonstantinS** [comment]
OnChainOrders have to be refactoring, now close this PR

**Pull request closed without merging**